### PR TITLE
#41 Remove Explicit Foreign Keys

### LINKED ISSUE #41: Remove Explicit Foreign Keys
**Problem**: Don't need to add an extra field for foreign keys
**Solution**: Remove explicit foreign keys

**Description**: Each relation has an Id for the two entities. For example, StudentExam has an id for the student and an id for the exam. We realized that when creating the relation, those ids we already there so we don't need to make a new field for them. Should remove these explicit foreign keys for every entity and make sure those Ids can still be accessed. 

**Specs:**
- [x] removed all explicit foreign keys
- [x] everything still works

If blocked, contact @nicolesorial  or @emaela

### PR DESCRIPTION
**Pull request by sophiacrennan**

### CODE REVIEW

**Review by emaela — CHANGES_REQUESTED** [review]
In ExamRoutes, the post route is missing an action, which should be "save". After adding that, everything worked as expected :)

File: src/entity/StudentCourse.ts [hunk 1]
```diff
@@ -35,9 +35,9 @@ export class StudentCourse {
...
     letterGrade: LetterGrade;
 
-    @ManyToOne(type => Student, student => student.studentCourses)
+    @ManyToOne(type => Student, student => student.studentCourses, {eager: true})
     student: Student;
 
-    @ManyToOne(type => Course, course => course.studentCourses)
+    @ManyToOne(type => Course, course => course.studentCourses, {eager: true})
     course: Course;
 }
```
Comment (nicolesorial) [inline src/entity/StudentCourse.ts]: Nit: new line

File: src/entity/StudentExam.ts [hunk 2]
```diff
@@ -47,10 +37,10 @@ export class StudentExam {
     })
     letterGrade: LetterGrade;
 
-    @ManyToOne(type => Student, student => student.studentExams)
+    @ManyToOne(type => Student, student => student.studentExams, {eager:true})
```
Comment (nicolesorial) [inline src/entity/StudentExam.ts:40]: I like what `eager: true` does in the HTTP request -- it isn't set for all the relations though (i.e., when I GET notes, I don't see what students they're related to). Maybe another ticket to do this for all entity relations?

File: src/entity/StudentExam.ts [hunk 3]
```diff
@@ -47,10 +37,10 @@ export class StudentExam {
...
     student: Student;
 
-    @ManyToOne(type => Exam, exam => exam.studentExams)
+    @ManyToOne(type => Exam, exam => exam.studentExams, {eager:true})
     exam: Exam;
 
 }
```
Comment (nicolesorial) [inline src/entity/StudentExam.ts]: Nit: new line

**Review by nicolesorial — APPROVED** [review]

**Pull request merged**